### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by Blackhex**
Updates the `README.ms` according to the current project status and adds a new chapter about testing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,33 +1,63 @@
...
 
-This repository contains a script to build a GNU toolchain cross compiler for Windows-on-arm64. It is a **work in progress**. This allows binaries that can execute on Windows-on-arm64 to be built on Linux. I have only tested this script on Ubuntu/WSL.
+This repository contains scripts for building a `x86_64-pc-linux-gnu` host, `aarch64-w64-mingw32` target GNU toolchain cross-compiler for Windows on Arm64. It is a **work in progress** in the stage of upstreaming the necessary changes to the corresponding repositories. This toolchain contains binaries that can be executed on Windows on Arm64 to be built on Linux. The scripts are being actively tested on the default Ubuntu 22.04 GitHub Actions runners and Ubuntu 22.04 in WSL.
```
Comment (eukarpov) [inline README.md:5]: I would rephrase it: for building the aarch64-w64-mingw32 target on aarch64-pc-linux-gnu or x86_64-pc-linux-gnu hosts.

File: README.md [hunk 2]
```diff
@@ -1,33 +1,63 @@
...
- - Downloads dependencies
- - Clones several dependencies 
-    - modified binutils from https://github.com/Windows-on-ARM-Experiments/binutils-woarm64.git
-    - modified gcc from https://github.com/Windows-on-ARM-Experiments/gcc-woarm64
-    - modified mingw from https://github.com/Windows-on-ARM-Experiments/mingw-woarm64
- - Builds and installs: binutils, mingw, gcc and libgcc.
+# Building the Cross-Compiler
 
-# Using the cross compiler
+Running [`./build.sh`](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/blob/main/build.sh) on Ubuntu 22.04 in the root repository folder will build the cross-compiler and install it into `~/cross-aarch64-w64-mingw32-msvcrt` folder by default.
```
Comment (eukarpov) [inline README.md:17]: For simplicity it would be good add step with cloning and entering the folder.

File: README.md [hunk 3]
```diff
@@ -1,33 +1,63 @@
...
+  - modified GCC from [Windows-on-ARM-Experiments/gcc-woarm64](https://github.com/Windows-on-ARM-Experiments/gcc-woarm64),
+  - modified MinGW from [Windows-on-ARM-Experiments/mingw-woarm64](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64).
+- Bootstraps the binutils and GCC sources with `gmp`, `mpfr`, `isl`, etc. dependencies.
+- Builds and installs binutils, stage 1 GCC, MinGW runtimes, GCC with libgcc, and then the entire MinGW.
+
+# Using the Cross-Compiler
+
+After building the toochain, to build a simple C source code file run:
+
+```bash
```
Comment (eukarpov) [inline README.md:43]: I would add enablement of the toolchain in PATH and using aarch64-w64-mingw32-gc without full path.

Comment (eukarpov) [reply README.md:43]: I would be also nice to describe how to build simple internal tests and run them.

File: README.md [hunk 4]
```diff
@@ -1,33 +1,63 @@
...
-# Install build dependencies:
+This toolchain is not yet ready for real-world use. Problems and missing parts are listed in the [issues](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/issues). Please, report all issues, even if the issue is, e.g., GCC-specific, here.
 
-sudo apt-get install build-essential binutils-for-build texinfo bison flex zlib1g-dev libgmp-dev dejagnu libmpc-dev
+# Installing Build Dependencies
 
-# Build the cross-compiler
+The [main build script](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/blob/main/build.sh) installs its dependencies automatically when `RUN_BOOTSTRAP=1` environment variable is defined which is the default. To see what will be installed refer to [`.github/scripts/install-dependencies.sh`](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/blob/main/.github/scripts/install-dependencies.sh).
 
-Running build.sh will build the cross compiler. It installs it into ~/cross. The build script does multiple things.
```
Comment (eukarpov) [inline README.md:16]: I think it would be useful to keep references to forks we are using. I suggest to move it to the bottom and correcting ~/cross folder.

Comment (Blackhex) [reply README.md:16]: I don't understand this suggestions? The references are still there it has just changed from git repository URLs to GitHub project URLs which is IMO more useful.

Comment (eukarpov) [reply README.md:16]: Thanks for clarification. For me it looked like it was removed.

File: README.md (hunk 1 above)
Comment (Blackhex) [reply README.md]: Rephrased.

File: README.md (hunk 3 above)
Comment (Blackhex) [reply README.md:76]: Changed to adding toolchain to `PATH`, added information on building the tests.

File: README.md (hunk 2 above)
Comment (Blackhex) [reply README.md]: Added steps with clonning.

File: README.md [hunk 5]
```diff
@@ -1,33 +1,103 @@
...
+[![Build main toolchain](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/actions/workflows/main.yml/badge.svg)](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/actions/workflows/main.yml) [![Build toolchain variants](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/actions/workflows/advanced.yml/badge.svg)](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/actions/workflows/advanced.yml) [![Daily rebase](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/actions/workflows/rebase.yml/badge.svg)](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/actions/workflows/rebase.yml)
 
-This repository contains a script to build a GNU toolchain cross compiler for Windows-on-arm64. It is a **work in progress**. This allows binaries that can execute on Windows-on-arm64 to be built on Linux. I have only tested this script on Ubuntu/WSL.
+This repository contains scripts for building an `aarch64-w64-mingw32` target Windows on Arm64
+GNU toolchain cross-compiler for `x86_64-pc-linux-gnu` hosts. It is a **work in progress**,
```
Comment (eukarpov) [inline README.md:6]: I would mention also aarch64-pc-linux-gnu.

File: README.md [hunk 6]
```diff
@@ -1,33 +1,103 @@
...
+  - modified MinGW from [Windows-on-ARM-Experiments/mingw-woarm64](https://github.com/Windows-on-ARM-Experiments/mingw-woarm64).
+- Bootstraps the binutils and GCC sources with `gmp`, `mpfr`, `isl`, etc. dependencies.
+- Builds and installs binutils, stage 1 GCC, MinGW runtimes, GCC with libgcc, and then
+  the entire MinGW.
+
+# Using the Cross-Compiler
+
+After building the toolchain, to build a simple C source code file run:
+```bash
+PATH="~/cross-aarch64-w64-mingw32-msvcrt/bin:$PATH" aarch64-w64-mingw32-gcc hello.c -o hello.exe
```
Comment (eukarpov) [inline README.md:72]: I think it is useful to enable the toolchain by extending PATH in a separate command.

Comment (Blackhex) [reply README.md]: Changed.

File: README.md (hunk 5 above)
Comment (Blackhex) [reply README.md]: Mention added.

**Review by eukarpov — APPROVED** [review]
